### ISSUE TITLE
[Tech Request]: performance regression after group function refactored

### ISSUE THREAD

**Issue by badboynt1** [matrixorigin/matrixone#20867]
### Is there an existing issue for the same tech request?

- [X] I have checked the existing issues.

### Does this tech request not affect user experience?

- [X] This tech request doesn't affect user experience.

### What would you like to be added ?

```Markdown
在聚合算子重构后，tpch性能下降。 主要是q1 / q15等等
```

### Additional information

![840cbaae33155f92e979077655fe3d2](https://github.com/user-attachments/assets/f843a9e0-c2e9-41af-b908-210151fe2448)

**Comment by fengttt** [comment]
@aressu1985 can you summarize this -- regression tracking, ...

**Comment by heni02** [comment]
confirm，closed
2.2 main分支性能已经恢复正常

<img width="932" alt="Image" src="https://github.com/user-attachments/assets/7559412c-70a1-49f5-bc1e-0ccc6eacabb4" />
